# HG changeset patch
# User sgehwolf
# Date 1589215018 -3600
#      Mon May 11 17:36:58 2020 +0100
# Node ID 39a7914e14a0c73ecc80091724c833b9f0410d47
# Parent  493a0176587ec5917ab4a64c0760a131e9d0b7b4
8233880: Support compilers with multi-digit major version numbers
Summary: Also includes JDK-8151841 version changes regressed by JDK-8034788
Reviewed-by: aph, andrew

diff --git openjdk.orig/common/autoconf/toolchain.m4 openjdk/common/autoconf/toolchain.m4
--- openjdk.orig/common/autoconf/toolchain.m4
+++ openjdk/common/autoconf/toolchain.m4
@@ -433,7 +433,7 @@
     COMPILER_VERSION_STRING=`$ECHO $COMPILER_VERSION_OUTPUT | \
         $SED -e 's/ *Copyright .*//'`
     COMPILER_VERSION_NUMBER=`$ECHO $COMPILER_VERSION_OUTPUT | \
-        $SED -e 's/^.* \(@<:@1-9@:>@\.@<:@0-9.@:>@*\) .*$/\1/'`
+        $SED -e 's/^.* \(@<:@1-9@:>@@<:@0-9@:>@*\.@<:@0-9.@:>@*\) .*$/\1/'`
   elif test  "x$TOOLCHAIN_TYPE" = xclang; then
     # clang --version output typically looks like
     #    Apple LLVM version 5.0 (clang-500.2.79) (based on LLVM 3.3svn)
